recipes/server_extend.rb

```diff
@@ -16,30 +16,32 @@
       Chef::Log.warn("Unable to find a chef node for #{peer}")
       next
     end
-    peer_bricks = chef_node['gluster']['server']['bricks'].select { |brick| brick.include? volume_name }
+    peer_bricks = chef_node['gluster']['server'][volume_name]['bricks'].select { |brick| brick.include? volume_name }
     brick_count += (peer_bricks.count || 0)
     peer_bricks.each do |brick|
       Chef::Log.info("Checking #{peer}:#{brick}")
       unless brick_in_volume?(peer, brick, volume_name)
-        node.default['gluster']['server']['bricks_waiting_to_join'] << " #{peer}:#{brick}"
+        node.default['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'] << " #{peer}:#{brick}"
       end
     end
   end
 
   replica_count = volume_values['replica_count']
-  next if node['gluster']['server']['bricks_waiting_to_join'].empty?
+  next if node['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'].empty?
```

Comment:
Chef run will bomb out here if the `node['gluster']['server']['volumes'][volume_name]['replica_count']` has been increased but the peer list hasn't been increased (i.e. `bricks_waiting_to_join`)

Think this is caused by https://github.com/shortdudey123/chef-gluster/pull/47/files#diff-3a2da8769b6e3177c6450e3d169fcdd7R58

`'bricks_waiting_to_join' = ''` should be namespaced to the volume name level, not the server level